### PR TITLE
Timezone map

### PR DESCRIPTION
**Pull request by jpnurmi**
A standalone timezone map widget, controller, and service.

Includes a small example app that looks a lot like https://github.com/canonical/ubuntu-desktop-installer/issues/38 :)
screenshot

### Credits

The SVG map assets are from [libtimezonemap](https://launchpad.net/ubuntu/+source/libtimezonemap).

### CODE REVIEW

**Comment by jpnurmi** [comment]
# macOS

image

**Comment by jpnurmi** [comment]
# Windows

image

File: packages/timezone_map/lib/src/map.dart [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    final exePath = Platform.resolvedExecutable;
+    final assetDir = fs.directory(p.join(p.dirname(exePath), 'data',
+        'flutter_assets', 'packages', 'timezone_map', 'assets'));
+    final assets = <String>[];
+    if (assetDir.existsSync()) {
+      assets.addAll(assetDir
+          .listSync()
+          .map((entity) => p.basename(entity.path))
+          .where((asset) => p.extension(asset) == '.svg'));
+    }
```
Comment (oSoMoN) [inline packages/timezone_map/lib/src/map.dart:48]: This method of listing all the svg assets within the package looks rather complicated. I've tested the following locally (only on the example app, not within the context of a third party app like the installer) and it seems to do the job with much less code:

```dart
DefaultAssetBundle.of(context)
    .loadString('AssetManifest.json')
    .then((value) => json.decode(value).keys.where((asset) =>
        p.isWithin('packages/timezone_map', asset) &&
        p.extension(asset) == '.svg'))
    .then(…)
```

**Review by oSoMoN — COMMENTED** [review]
LGTM! Just one suggestion on how to maybe simplify SVG assets preloading.

File: packages/timezone_map/lib/src/map.dart (hunk 1 above)
Comment (jpnurmi) [reply packages/timezone_map/lib/src/map.dart:48]: much nicer, thanks! :+1: i'll give it a try and update the tests

**Pull request merged**